Compiler/BackEnd/FindZeroCrossings.mo

```diff
@@ -176,20 +176,27 @@ algorithm
     then (eqn, (equationArray, vars, eqns, index, ht));
 
     // removed algorithm
-    case (BackendDAE.ALGORITHM(size=0, alg=alg_, source=source, expand=crefExpand, attr=attr), (equationArray, vars, eqns, index, ht)) equation
-      DAE.ALGORITHM_STMTS(statementLst=stmts) = alg_;
-      size = -index;
-      (stmts, preStmts, index) = encapsulateWhenConditions_Algorithms(stmts, vars, index);
-      sizePre = listLength(preStmts);
-      size = size+index-sizePre;
-
-      alg_ = DAE.ALGORITHM_STMTS(stmts);
-      eqn = BackendDAE.ALGORITHM(size, alg_, source, crefExpand, attr);
-      equationArray = BackendEquation.addEquation(eqn, equationArray);
+    case (BackendDAE.ALGORITHM(size=0, alg=alg_, source=source, expand=crefExpand, attr=attr), (equationArray, vars, eqns, index, ht)) algorithm
+      DAE.ALGORITHM_STMTS(statementLst=stmts) := alg_;
+      size := -index;
+      allPreStmts := {};
+      allStmts := {};
+      for stmt in stmts loop
+        (stmts, preStmts, index) := encapsulateWhenConditions_Algorithms({stmt}, vars, index);
```

Comment:
If coverage is fine, I will check if I can simplify encapsulateWhenConditions_Algorithms furthermore